coderd/rbac/policy.rego

```diff
@@ -93,6 +93,8 @@ default scope_org := 0
 scope_org := org_allow([input.scope])
 
 org_allow(roles) := num {
+	# Rule index on org_owner being set.
+	input.object.org_owner != ""
```

Comment:
[Indexed Statement](https://www.openpolicyagent.org/docs/latest/policy-performance/#use-indexed-statements)

We already do this on user_allow. This saves allocations on authorize if the object has no org owners. It adds a small amount to partial execution by adding an extra query expression. See below.

# Before (with no org owner)

```
+-------------------------------------------+------------+
| samples                                   |       2318 |
| ns/op                                     |     504203 |
| B/op                                      |     140009 |
| allocs/op                                 |       2733 |
| histogram_timer_rego_query_eval_ns_75%    |     524535 |
| histogram_timer_rego_query_eval_ns_90%    |     675357 |
| histogram_timer_rego_query_eval_ns_95%    |    1010606 |
| histogram_timer_rego_query_eval_ns_99%    |    1333671 |
| histogram_timer_rego_query_eval_ns_99.9%  |    1661102 |
| histogram_timer_rego_query_eval_ns_99.99% |    1664193 |
| histogram_timer_rego_query_eval_ns_count  |       2318 |
| histogram_timer_rego_query_eval_ns_max    |    1664193 |
| histogram_timer_rego_query_eval_ns_mean   |     494290 |
| histogram_timer_rego_query_eval_ns_median |     477631 |
| histogram_timer_rego_query_eval_ns_min    |     294300 |
| histogram_timer_rego_query_eval_ns_stddev |     210656 |
+-------------------------------------------+------------+
```
# After (no org set)

```
+-------------------------------------------+------------+
| samples                                   |       2967 |
| ns/op                                     |     380418 |
| B/op                                      |     117387 |
| allocs/op                                 |       2197 |
| histogram_timer_rego_query_eval_ns_75%    |     395319 |
| histogram_timer_rego_query_eval_ns_90%    |     545068 |
| histogram_timer_rego_query_eval_ns_95%    |     761163 |
| histogram_timer_rego_query_eval_ns_99%    |    1092210 |
| histogram_timer_rego_query_eval_ns_99.9%  |    1225354 |
| histogram_timer_rego_query_eval_ns_99.99% |    1225792 |
| histogram_timer_rego_query_eval_ns_count  |       2967 |
| histogram_timer_rego_query_eval_ns_max    |    1225792 |
| histogram_timer_rego_query_eval_ns_mean   |     372498 |
| histogram_timer_rego_query_eval_ns_median |     355547 |
| histogram_timer_rego_query_eval_ns_min    |     216730 |
| histogram_timer_rego_query_eval_ns_stddev |     172728 |
+-------------------------------------------+------------+
```

# Extra partial query

Adds redundent `input.object.org_owner != ""`. We could optimize this in the `CompileToSQL` if we cannot figure out how to do it in the rego after being partially compiled.

```
+---------+---------------------------------------------------------------------------------------------+
| Query 2 | input.object.org_owner != ""                                                                |
|         | "bf7b72bd-a2b1-4ef2-962c-1d698e0483f6" = input.object.org_owner                             |
+---------+---------------------------------------------------------------------------------------------+
```

Reply:
Ah dang. This causes a supporting section which is not allowed in a partial query in a unit test. Will continue looking into this.